logic.py

```diff
@@ -1415,6 +1417,33 @@ def subst(s, x):
     else:
         return Expr(x.op, *[subst(s, arg) for arg in x.args])
 
+def cascade_substitution(s):
+    '''This method allow to return a correct unifier in normal form
```

Comment:
In the repo we generally use double quotes for docstrings. I would prefer it that way for consistency's sake.

Also, it should be 'allows' instead of 'allow'.